### PR TITLE
Kernel: Make the ACPI DSDT table accessible from ACPI::Parser and in /sys/firmware/acpi

### PR DESCRIPTION
**Pull request by me-minus**
A first little step toward interpreting the AML bytecode from ACPI.

### CODE REVIEW

File: Kernel/Firmware/ACPI/Parser.cpp [hunk 1]
```diff
@@ -205,6 +206,19 @@ UNMAP_AFTER_INIT void Parser::process_fadt_data()
     m_hardware_flags.wbinvd_flush = (sdt->flags & (u32)FADTFlags::FeatureFlags::WBINVD_FLUSH);
 }
 
+UNMAP_AFTER_INIT void Parser::process_dsdt()
+{
+    auto sdt = Memory::map_typed<Structures::FADT>(m_fadt).release_value_but_fixme_should_propagate_errors();
+
+    m_sdt_pointers.append(PhysicalAddress(sdt->dsdt_ptr));
+    auto dsdt_or_error = Memory::map_typed<Structures::DSDT>(PhysicalAddress(sdt->dsdt_ptr));
```
Comment (supercomputer7) [inline Kernel/Firmware/ACPI/Parser.cpp:214]: What about this line?
https://github.com/SerenityOS/serenity/blob/a1943039489b1e9c4742498ee8424e10379de5a4/Kernel/Firmware/ACPI/Parser.cpp#L173

Comment (me-minus) [reply Kernel/Firmware/ACPI/Parser.cpp:214]: > What about this line?
> 
> https://github.com/SerenityOS/serenity/blob/a1943039489b1e9c4742498ee8424e10379de5a4/Kernel/Firmware/ACPI/Parser.cpp#L173

Other ACPI tables print their physical address so I did the same

Comment (supercomputer7) [reply Kernel/Firmware/ACPI/Parser.cpp:214]: What I mean is that it will print twice the same line for no good reason.

**Comment by supercomputer7** [comment]
The commit message is misleading in this part:
```
in /sys/firmware/acpi
```
How it exposes the DSDT table now? I don't see how it happens with your patch.

**Comment by me-minus** [comment]
> The commit message is misleading in this part:
> 
> ```
> in /sys/firmware/acpi
> ```
> 
> How it exposes the DSDT table now? I don't see how it happens with your patch.

line 213 in Parser.cpp:
m_sdt_pointers.append(PhysicalAddress(sdt->dsdt_ptr));

m_sdt_pointer is later exposed as files, with names from the ACPI table header structure of the pointers.
See  Parser::enumerate_static_tables.

File: Kernel/Firmware/ACPI/Parser.cpp (hunk 1 above)
Comment (me-minus) [reply Kernel/Firmware/ACPI/Parser.cpp:213]: I removed the printout in process_fadt_data and extended the message in process_dsdt

**Comment by supercomputer7** [comment]
> > The commit message is misleading in this part:
> > ```
> > in /sys/firmware/acpi
> > ```
> >   
> > How it exposes the DSDT table now? I don't see how it happens with your patch.
> 
> line 213 in Parser.cpp: m_sdt_pointers.append(PhysicalAddress(sdt->dsdt_ptr));
> 
> m_sdt_pointer is later exposed as files, with names from the ACPI table header structure of the pointers. See Parser::enumerate_static_tables.

I see, please add a comment on line 213 to ensure we know this :)
Something like this will suffice:
```
// Note: We add the DSDT to the m_sdt_pointers Vector so later on it is enumerated and the DSDT will be exposed
// in the SysFS under the "firmware/acpi" path.
```

**Comment by me-minus** [comment]
> > > The commit message is misleading in this part:
> > > ```
> > > in /sys/firmware/acpi
> > > ```
> > > 
> > > 
> > >     
> > >       
> > >     
> > > 
> > >       
> > >     
> > > 
> > >     
> > >   
> > > How it exposes the DSDT table now? I don't see how it happens with your patch.
> > 
> > 
> > line 213 in Parser.cpp: m_sdt_pointers.append(PhysicalAddress(sdt->dsdt_ptr));
> > m_sdt_pointer is later exposed as files, with names from the ACPI table header structure of the pointers. See Parser::enumerate_static_tables.
> 
> I see, please add a comment on line 213 to ensure we know this :) Something like this will suffice:
> 
> ```
> // Note: We add the DSDT to the m_sdt_pointers Vector so later on it is enumerated and the DSDT will be exposed
> // in the SysFS under the "firmware/acpi" path.
> 

Added a comment with a slightly shorter message

**Pull request merged**